### PR TITLE
DON'T MERGE!!! Make default behavior of "Publish"/"Publish all" buttons configurable #107

### PR DESCRIPTION
**Pull request by danielkestler**
Add a user preference ```publishingModeAll``` (true/false) to configure the publishing behavior, see: https://github.com/neos/neos-ui/issues/107

If set to true, the main publish dropdown button will publish all dirty nodes, if set to false, only the current document node.

Also fixes a bug (?): the auto-publishing settings have not been saved into user preferences until now, this is also fixed.

bildschirmfoto 2018-10-26 um 15 05 22

### CODE REVIEW

**Comment by markusguenther** [comment]
Nice feature. We have some conflicts in the redux store. As I am not permitted to adjust that, it would be nice if you can resolve the conflicts :)

**Comment by danielkestler** [comment]
I tried to migrate it to Typescript – hope I did it right, as I have no experience using Typescript yet ;-)

**Comment by danielkestler** [comment]
Think I'm having it. Let's see what Travis says.

**Comment by markusguenther** [comment]
Awesome :) 

You can also test it locally with „make test“ and „make lint“

Wie have a List of the make commands in the global readme. In the single package you can also execute jest or yarn lint

**Comment by danielkestler** [comment]
Sorry, I thought I'm having it but I made a small mistake. Be right back ;-) Testing it locally through the command line (forgot one test ... it is not self explanatory if you are "new" to this JS world).

**Comment by markusguenther** [comment]
This is nothing you have to complain about. We are all people that learning new technologies and know that we can’t be experts in all fields. I appreciate that you push this thing and that your not afraid of the JavaScript part :)

File: packages/neos-ui-redux-store/src/User/Settings/index.ts [hunk 1]
```diff
@@ -43,10 +47,14 @@ export const reducer = (state: State = defaultState, action: Action | InitAction
     switch (action.type) {
         case system.INIT:
             draft.isAutoPublishingEnabled = action.payload.user.settings.isAutoPublishingEnabled;
```
Comment (mstruebing) [inline packages/neos-ui-redux-store/src/User/Settings/index.ts]: The error seems to be here, look at: https://travis-ci.org/neos/neos-ui/jobs/469367733#L1543

**Comment by danielkestler** [comment]
Yeah, there was one completely wrong test (maybe copy and paste error on my side) that lead to the fail and one other test that in itself did not make sense.

**Comment by danielkestler** [comment]
Merged master and resolved conflicts, now the two settings are no longer correctly saved. Tried debugging it, and it seems the action.payload of the sagas is always "undefined". Worked before ...

**Comment by Sebobo** [comment]
Hi @danielkestler, is this still broken, or can it be tested?

**Comment by danielkestler** [comment]
@Sebobo Should be working now, the UserState was missing. The tests fail because this PR should maybe be rebased on 2.x, but I completely messed up last time ... so I probably won't do that again ;-)

**Comment by dimaip** [comment]
@danielkestler hey, do you still want to give fixing the tests a try? As you said a rebase should help.

**Comment by danielkestler** [comment]
@dimapi The tests fail because the testing is on CircleCI now ... I know a rebase should help, but as I completely messed up a rebase last time, I don't know if I should try again. Maybe you could do it or just tell me the steps you usually perform to make a rebase ...? Thank you very much :-)

**Comment by dimaip** [comment]
@danielkestler unfortunately I'm super limited on time lately so can't dive deep into it.
I suck at kick too, so if I screw up, I do the following:
1. note create a new branch (based on 2.x in this case)
2. cherry-pick commits from the old branch

In this branch you have a lot of commits, so I would also squash them first (`git rebase -i HEAD~20`)

**Comment by danielkestler** [comment]
@dimaip Thank you, I will try. Will I then have to make a new PR or do I then merge the new branch in this one here?

**Comment by dimaip** [comment]
You could either create a new PR or force-push to this branch

**Comment by danielkestler** [comment]
Tried and messed up again. Cherry-picking went quite fine I think, but force pushing leads to results I don't understand. Made a new PR: https://github.com/neos/neos-ui/pull/2549

**Pull request closed without merging**